pkg/sql/colexec/generate_series/generate_series_test.go

```diff
@@ -0,0 +1,495 @@
+			step:  "1 year",
+			res: []types.Datetime{
+				transStr2Datetime("2020-02-29 00:01:00"),
+				transStr2Datetime("2021-02-28 00:01:00"),
+			},
+		},
+		{
+			start: "2020-02-28 00:01:00",
+			end:   "2021-03-01 00:00:00",
+			step:  "1 year",
+			res: []types.Datetime{
+				transStr2Datetime("2020-02-28 00:01:00"),
+				transStr2Datetime("2021-02-28 00:01:00"),
+			},
+		},
+	}
+	for _, kase := range kases {
+		var precision int32
+		p1, p2 := getPrecision(kase.start), getPrecision(kase.end)
+		if p1 > p2 {
+			precision = p1
+		} else {
+			precision = p2
+		}
+		res, err := generateDatetime(kase.start, kase.end, kase.step, precision)
+		if kase.err {
+			require.NotNil(t, err)
+			continue
+		}
+		require.Nil(t, err)
+		require.Equal(t, kase.res, res)
+	}
+}
+
+func transStr2Datetime(s string) types.Datetime {
+	precision := getPrecision(s)
+	t, err := types.ParseDatetime(s, precision)
+	if err != nil {
+		logutil.Errorf("parse timestamp '%s' failed", s)
+	}
+	return t
+}
+
+func getPrecision(s string) int32 {
+	var precision int32
+	ss := strings.Split(s, ".")
+	if len(ss) > 1 {
+		precision = int32(len(ss[1]))
+	}
+	return precision
+}
+
+func TestString(t *testing.T) {
+	String(nil, new(bytes.Buffer))
+}
+
+func TestPrepare(t *testing.T) {
+	err := Prepare(nil, new(bytes.Buffer))
+	require.Nil(t, err)
+}
+func TestGenStep(t *testing.T) {
+	kase := "10 hour"
+	num, tp, err := genStep(kase)
+	require.Nil(t, err)
+	require.Equal(t, int64(10), num)
+	require.Equal(t, types.Hour, tp)
+	kase = "10 houx"
+	_, _, err = genStep(kase)
+	require.NotNil(t, err)
+	kase = "hour"
+	_, _, err = genStep(kase)
+	require.NotNil(t, err)
+	kase = "989829829129131939147193 hour"
+	_, _, err = genStep(kase)
+	require.NotNil(t, err)
+}
+
+func TestCall(t *testing.T) {
+	proc := testutil.NewProc()
+	arg := &Argument{
+		Es: &Param{
+			Attrs: []string{"result"},
+			Cols: []*plan.ColDef{
+				{
+					Name: "result",
+					Typ:  &plan.Type{},
+				},
+			},
+		},
+	}
+	param := arg.Es
+	proc.SetInputBatch(nil)
+	end, err := Call(0, proc, arg)
+	require.Nil(t, err)
+	require.Equal(t, true, end)
+
+	param.Cols[0].Typ.Id = int32(types.T_int32)
+	proc.SetInputBatch(makeBatch([]string{"1", "10"}, proc))
+	end, err = Call(0, proc, arg)
+	require.Nil(t, err)
+	require.Equal(t, false, end)
+	require.Equal(t, 10, proc.InputBatch().GetVector(0).Length())
+
+	param.Cols[0].Typ.Id = int32(types.T_int64)
+	proc.SetInputBatch(makeBatch([]string{"654345676543", "654345676549"}, proc))
+	end, err = Call(0, proc, arg)
+	require.Nil(t, err)
+	require.Equal(t, false, end)
+	require.Equal(t, 7, proc.InputBatch().GetVector(0).Length())
+
+	param.Cols[0].Typ.Id = int32(types.T_datetime)
+	param.Cols[0].Typ.Precision = 0
+	proc.SetInputBatch(makeBatch([]string{"2020-01-01 00:00:00", "2020-01-01 00:00:59", "1 second"}, proc))
+	end, err = Call(0, proc, arg)
+	require.Nil(t, err)
+	require.Equal(t, false, end)
+	require.Equal(t, 60, proc.InputBatch().GetVector(0).Length())
```

Comment:
clean the inputBatch and check if (mPool.Now() equals to mPool.BeforeCall())

Reply:
OK

Reply:
> 

What does `mPool.Now() equals to mPool.BeforeCall()` means? I cannot find functions like these.